include/RAJA/policy/cuda/atomic.hpp

```diff
@@ -590,177 +602,162 @@ struct cuda_atomic {
  * These are atomic in cuda device code and non-atomic otherwise
  */
 RAJA_SUPPRESS_HD_WARN
-template <typename T>
-RAJA_INLINE RAJA_HOST_DEVICE T atomicAdd(cuda_atomic, T volatile *acc, T value)
+template <typename T, typename host_policy>
+RAJA_INLINE RAJA_HOST_DEVICE T
+atomicAdd(cuda_atomic_explicit<host_policy>, T volatile *acc, T value)
 {
 #ifdef __CUDA_ARCH__
   return detail::cuda_atomicAdd(acc, value);
 #else
-  T old = *acc;
-  *acc = old + value;
-  return old;
+  return atomicAdd(host_policy{}, acc, value);
```

Comment:
Everything looks good. These calls are coming from loop_atomic or seq_atomic? If so, maybe we should add a note for clarity. I was going to suggest adding more scoping resolution for these calls, but it seems like they reside in the `RAJA::` namespace.

Reply:
They can come from any host atomic policy like loop_atomic or seq_atomic. I didn't add more namespace scoping because we directly used detail:: above instead of RAJA::detail, but that's not a bad idea.